core/src/lib.rs

```diff
@@ -14,23 +23,36 @@ pub trait DIDResolver {
     ) -> Result<(), Box<dyn std::error::Error>>;
 }
 
-/// Given the credential type and the credential subject information, create a unissued JSON-LD credential.
-/// In order to become a Verifiable Credential, a data integrity proof must be created for the credential and appended to the JSON-LD document.
-pub fn create_credential(
-    _cred_type: &str,
-    _cred_subject: serde_json::Value,
-) -> Result<serde_json::Value, Box<dyn std::error::Error>> {
-    unimplemented!();
-}
+pub trait DocumentBuilder {
+    /// Given the credential type and the credential subject information, create a unissued JSON-LD credential.
+    /// In order to become a Verifiable Credential, a data integrity proof must be created for the credential and appended to the JSON-LD document.
+    /// this is the default implementation of the `create` method. The `create` method can be overridden to create a custom credential.
+    pub fn create_credential(
+        &self,
+        cred_type: String, 
+        cred_subject: HashMap<String, Value>,
+        property_set: HashMap<String, Value>,
+        id: &str
+    ) -> Result<serde_json::Value, Box<dyn std::error::Error>> {
```

Comment:
Also here we should probably just return `vc` instead of the `serde_json::Value` serialization. We should let the user decide if they want to serialize it. 

Since we're so level we would want to make as few decisions for the user as possible and serialization is a big enough deal to be hesitant to do so. Also the user may need to do some things with the VC afterwards and JSON is harder to deal with than an actual struct

Reply:
Yes, completely agree, it is always simpler to handle objects. Our first interface was indeed JSON but know it seems to be evident that objects are more suitable here. Moreover later we can accept this object as an input for subsequent calls